src/platform/packages/shared/kbn-apm-synthtrace/src/cli/utils/stream_manager.ts

```diff
@@ -98,24 +98,38 @@ export class StreamManager {
     generator: SynthGenerator<Fields>
   ): Promise<void> {
     const clientStream = this.createOrReuseClientStream(client);
+    const generatorStream = new PassThrough({ objectMode: true });
+    const generatorArray = castArray(generator);
+    const streams: Readable[] = [];
+
+    generatorArray.reverse().forEach((current) => {
+      const currentStream = isGeneratorObject(current) ? Readable.from(current) : current;
+      currentStream.pipe(generatorStream, { end: false });
+      streams.push(currentStream);
+
+      currentStream.on('end', () => {
+        pull(streams, currentStream);
+        if (streams.length === 0) {
+          generatorStream.end();
+        }
+      });
 
-    const generatorStream = castArray(generator)
-      .reverse()
-      .reduce<Writable>((prev, current) => {
-        const currentStream = isGeneratorObject(current) ? Readable.from(current) : current;
-        return currentStream.pipe(prev);
-      }, new PassThrough({ objectMode: true }));
+      currentStream.on('error', (err) => {
+        pull(streams, currentStream);
+        generatorStream.destroy(err);
```

Comment:
should this also destroy other streams?

Reply:
I think so. [90df3eb](https://github.com/elastic/kibana/pull/215291/commits/90df3ebbec667888ac40feff69aa4ae39cf02f06)